### DISCUSSION TITLE
Direct Messages Cursor(pagination)

### DISCUSSION THREAD

**Discussion by manojbanoth** [tweepy/tweepy#1897] [Questions] [answered]
tweepy version ==4.9.0

api.get_direct_messages(count=500 , cursor=-1) with this code gave me an error Invalid cursor

tweepy.Cursor(api.get_direct_messages(count=100)) with this code gave me an error ('This method does not perform pagination')
api.get_direct_messages(count=5000) gives me only last 50 messages 

But the document says pagination is possible.  
[https://developer.twitter.com/en/docs/twitter-api/v1/direct-messages/sending-and-receiving/api-reference/list-events]

I want to extract all the DM for the past 30 days within the rate limit 
please do needful

**Comment by Harmon758** [comment, marked as answer]
With `tweepy.Cursor(api.get_direct_messages(count=100))`, you're passing the results of your call to `API.get_direct_messages` to `Cursor`. You need to pass the method itself.

See [the documentation on pagination using `Cursor`](https://tweepy.readthedocs.io/en/v4.9.0/v1_pagination.html) for more information.

**Reply by manojbanoth** [reply]
@Harmon758 Thanks for giving the best suitable answer. It worked charmingly.
